runtime/runtime/src/verifier.rs

```diff
@@ -778,61 +753,49 @@ mod tests {
+        /// Testing all combination of access keys in this test to make sure that an account
+        /// is zero balance only if it uses <= `ZERO_BALANCE_ACCOUNT_STORAGE_LIMIT` bytes in storage
         #[test]
-        fn test_zero_balance_account_with_keys_and_contract() {
+        fn test_zero_balance_account_with_keys() {
             for num_full_access_key in 0..10 {
                 for num_function_call_access_key in 0..10 {
-                    for i in 0..=1 {
-                        for j in 0..=1 {
-                            let account_id: AccountId = format!(
-                                "alice{}.near",
-                                num_full_access_key * 1000
-                                    + num_function_call_access_key * 100
-                                    + i * 10
-                                    + j
-                            )
-                            .parse()
-                            .unwrap();
-                            let has_contract_code = i == 0;
-                            let has_contract_data = j == 0;
-                            let (account, mut state_update) = set_up_test_account(
-                                &account_id,
-                                num_full_access_key,
-                                num_function_call_access_key,
-                                has_contract_code,
-                                has_contract_data,
-                            );
-                            let res =
-                                is_zero_balance_account(&account_id, &account, &mut state_update);
-                            if num_full_access_key <= 2
-                                && num_function_call_access_key <= 2
-                                && !has_contract_code
-                                && !has_contract_data
-                            {
-                                assert_eq!(res, Ok(true));
-                            } else {
-                                assert_eq!(res, Ok(false));
-                            }
-                        }
-                    }
+                    let account_id: AccountId = format!(
+                        "alice{}.near",
+                        num_full_access_key * 1000 + num_function_call_access_key
+                    )
+                    .parse()
+                    .unwrap();
+                    let (account, _) = set_up_test_account(
+                        &account_id,
+                        num_full_access_key,
+                        num_function_call_access_key,
+                    );
+                    let res = is_zero_balance_account(&account);
+                    assert_eq!(
+                        res,
+                        num_full_access_key * 82
+                            + num_function_call_access_key * 171
+                            + std::mem::size_of::<Account>() as u64
```

Comment:
I think we should hardcode 100 here to make it consistent with the formula - we don't measure Account size, we just use 100 from config. It's a bit suspicious that test pass anyway.

Reply:
it is actually 72 here because I believe it doesn't use the proper config in the test setup